bot/src/main/java/com/community/tools/service/github/PullRequestValidationService.java

```diff
@@ -0,0 +1,161 @@
+package com.community.tools.service.github;
+
+import com.community.tools.dto.CommitFileDto;
+import com.community.tools.dto.OpenAiValidationResponseDto;
+import com.community.tools.dto.QueuedValidationProcessDto;
+import com.community.tools.service.openai.OpenAiService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHPullRequestCommitDetail;
+import org.kohsuke.github.GHPullRequestReviewBuilder;
+import org.kohsuke.github.GHPullRequestReviewEvent;
+import org.kohsuke.github.GHRepository;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class PullRequestValidationService {
+  private final GitHubConnectService gitService;
+  private final ObjectMapper objectMapper;
+  private final OpenAiService openAiService;
+  private final Queue<QueuedValidationProcessDto> requests = new LinkedList<>();
+
+  /**
+   * Constructor for the service.
+   *
+   * @param gitService git service to get repositories
+   * @param objectMapper mapper to map responses to desired object
+   * @param openAiService ai service to validate PRs
+   */
+  public PullRequestValidationService(GitHubConnectService gitService,
+                                      ObjectMapper objectMapper, OpenAiService openAiService) {
+    this.gitService = gitService;
+    this.objectMapper = objectMapper;
+    this.openAiService = openAiService;
+  }
+
+  /**
+   * Main method of the service that retrieves Feedback PR by its id(1),
+   * iterates through its commits(except those by bot created)
+   * and gathers all commit files into one prompt to put it in the queue of validation.
+   *
+   * @param repositoryName name of the repository that needs its PR to be validated.
+   * @throws RuntimeException If github api doesn't find requested objects
+   */
+  public void validatePullRequest(String repositoryName) {
+    GHRepository repository = gitService.getGitHubRepositoryByName(repositoryName);
+    try {
+      GHPullRequest pullRequest = repository.getPullRequest(1);
+
+      for (GHPullRequestCommitDetail commitDetail :
+              pullRequest.listCommits().toList()
+                      .stream().filter(commit -> !commit.getCommit()
+                              .getAuthor().getName().equals("github-classroom[bot]"))
```

Comment:
Extract please those huge parts of code to separate variables, and format the identation to make it bit more readable

Reply:
Extracted lists to separate variables